src/extension/providers/debug_config_provider.ts

```diff
@@ -150,7 +150,7 @@ export class DebugConfigProvider implements DebugConfigurationProvider {
 			&& !isInsideFolderNamed(debugConfig.program, "tool")
 			&& !isInsideFolderNamed(debugConfig.program, ".dart_tool")) {
 			// Check if we're a Flutter or Web project.
-			if (isFlutterProjectFolder(debugConfig.cwd)) {
+			if (isFlutterProjectFolder(debugConfig.cwd) || this.wsContext.config.forceFlutterMode) {
```

Comment:
With this change, the `forceFlutterMode` flag is doing now doing two things:

- Forcing the _workspace_ into Flutter mode (this activates the daemon, enables Flutter commands)
- Forcing any debug session to use the Flutter debug adapter

These are subtly different, since a Flutter workspace doesn't usually mean the Flutter debugger would always be used (for example if you run a `tool/foo.dart` in a Flutter project, normally it would run with the Dart debugger - but this will cause it to run with Flutter).

I suspect that's intended, though it's worth adding a comment to `forceFlutterMode` to indicate this behaviour (or splitting it into two flags, one for forcing the workspace mode, and one for forcing the debug type) to make it clear.

Reply:
I see, I think it makes sense to split it up here then. I may eventually want to add some other checks for whether to use Flutter mode during debug.